include/z64save.h

```diff
@@ -245,6 +245,13 @@ typedef enum {
     /* 3 */ SUNSSONG_SPECIAL // time does not advance, but signals the song was played. used for freezing redeads
 } SunsSongState;
 
+typedef enum {
+    /* 0 */ GAMEMODE_NORMAL,
+    /* 1 */ GAMEMODE_TITLE_SCREEN,
+    /* 2 */ GAMEMODE_FILE_SELECT, // Note: only instance type transitions swap to file select
```

Comment:
What's an instance type transition?

Reply:
(copying from discord)

transitions start by setting `transitionTrigger`, `nextEntranceIndex`, `transitionType`, for example:
https://github.com/zeldaret/oot/blob/e16779cc338c53b91387a28e3e97cd6f5f036a66/src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c#L1498-L1500

they start being handled in z_play with the check on `transitionTrigger`:
https://github.com/zeldaret/oot/blob/e16779cc338c53b91387a28e3e97cd6f5f036a66/src/code/z_play.c#L481-L483

then a bunch of lines below, the transition type is passed to `Play_SetupTransition`
https://github.com/zeldaret/oot/blob/e16779cc338c53b91387a28e3e97cd6f5f036a66/src/code/z_play.c#L531

In `Play_SetupTransition`,
For instance type transitions, `transitionMode` isn't changed so stays `TRANS_MODE_SETUP`
for non-instance types, `transitionMode` is set accordingly

when `Play_SetupTransition` returns, the following block breaks the switch for non-instance modes
https://github.com/zeldaret/oot/blob/e16779cc338c53b91387a28e3e97cd6f5f036a66/src/code/z_play.c#L536-L539
but instance modes continue

eventually
https://github.com/zeldaret/oot/blob/e16779cc338c53b91387a28e3e97cd6f5f036a66/src/code/z_play.c#L630-L639
is reached as part of processing instance modes and that's where file select is switched to